diff --git a/dist/_internal/index.js b/dist/_internal/index.js
index 45d614184f271be0fee050610619e4f64b7961ae..ae8ef9fde9b7314401b348d38041b5f04b54332d 100644
--- a/dist/_internal/index.js
+++ b/dist/_internal/index.js
@@ -106,7 +106,7 @@ const createCacheHelper = (cache, key)=>{
             }
         },
         // Subscriber
-        state[6],
+        state && state[6] ? state[6] : () => {},
         // Get server cache snapshot
         ()=>{
             if (!isUndefined(key)) {
