diff --git a/package.json b/package.json
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
     "typescript": "^5.4.3"
   },
   "dependencies": {
-    "zod": "^3.22.4"
+    "zod": "^4.0.0"
   },
   "scripts": {
     "test": "node --loader=tsx --test ./**/*.test.ts ",
diff --git a/src/client.ts b/src/client.ts
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/src/client.ts
+++ b/src/client.ts
@@ -80,7 +80,7 @@ export class Tinybird {
   }): (
     params: z.input<TParameters>,
   ) => Promise<z.infer<typeof pipeResponseWithoutData> & { data: z.output<TData>[] }> {
-    const outputSchema = pipeResponseWithoutData.setKey("data", z.array(req.data));
+    const outputSchema = pipeResponseWithoutData.extend({ data: z.array(req.data) });
     return async (params: z.input<TParameters>) => {
       let validatedParams: z.input<TParameters> | undefined = undefined;
       if (req.parameters) {
