diff --git a/dist/shared/better-auth.CjNhg7P4.mjs b/dist/shared/better-auth.CjNhg7P4.mjs
index 4a8d8eb763db277adc589fa389808926fc6dedcb..273bd0cd8103f0a44b5c65e9f8de74a14c92b7b5 100644
--- a/dist/shared/better-auth.CjNhg7P4.mjs
+++ b/dist/shared/better-auth.CjNhg7P4.mjs
@@ -172,6 +172,7 @@ async function setCookieCache(ctx, session, dontRememberMe) {
       },
       {}
     );
+    session.user.image = null;
     const sessionData = { session: filteredSession, user: session.user };
     const options = {
       ...ctx.context.authCookies.sessionData.options,
